internal/contracts/manager.go

```diff
@@ -480,6 +480,14 @@ func (cm *contractManager) AddContractSubscription(ctx context.Context, ns strin
 			sub.Event = &fftypes.FFISerializedEvent{
 				FFIEventDefinition: event.FFIEventDefinition,
 			}
+		} else if sub.Event.Name != "" && sub.Interface != nil {
```

Comment:
What about other combinations, such as `sub.Event.Name == "" && sub.Interface != nil` - is that an error combination handled in `UpsertContractListener`?

Reply:
I did a little more testing on this to try different combinations of these things, and tweaked this just a bit. The case you were asking about was already covered by the `validateFFIEvent` function. I added some additional tests as well. I think we're covered here now.